### PR TITLE
chore: Update README.md to add links to each sections

### PR DESCRIPTION
**Pull request by Musaddiq625**
## What kind of change does this PR introduce?

Updated README.md doc

## What is the current behavior?

For now the user have to read the whole document to find all the features

## What is the new behavior?

Added list of features on top of the document for better understanding

### CODE REVIEW

**Review by dshukertjr — COMMENTED** [review]
Thanks for this. Could we make them links to each sections?

**Comment by Musaddiq625** [comment]
yes I was trying it
but the linking wasn't working in the 'preview'
let me try again 🔁

**Comment by Musaddiq625** [comment]
@dshukertjr 
Please do check 🙌🏻

File: packages/supabase_flutter/README.md [hunk 1]
```diff
@@ -39,7 +39,22 @@ final supabase = Supabase.instance.client;
 
 ## Usage example
 
-### [Authentication](https://supabase.com/docs/guides/auth)
+* [Authentication](#authentication)
+* [Native Apple Sign in](#native-apple-sign-in)
+* [Native Google sign in](#native-google-sign-in)
+* [OAuth login](#oauth-login)
```
Comment (dshukertjr) [inline packages/supabase_flutter/README.md:45]:
```suggestion
* [Authentication](#authentication)
  * [Native Apple Sign in](#native-apple-sign-in)
  * [Native Google sign in](#native-google-sign-in)
  * [OAuth login](#oauth-login)
```

**Review by dshukertjr — APPROVED** [review]
Looks good! Thanks for the contributions!

**Comment by Musaddiq625** [comment]
Hello @dshukertjr 
I have some questions on this:
Is this the same README file on [Supabase pub.dev](https://pub.dev/packages/supabase_flutter)?
and if not, how this will be updated on pub dev (as most of the flutter devs read the README directly from the pub.dev)
thankss

**Comment by Vinzent03** [comment]
It will get updated on pub.dev as soon as we release a new version.

**Comment by Musaddiq625** [comment]
got it, thanks 🙌🏻

**Comment by Musaddiq625** [comment]
got it, thanks @Vinzent03  🙌🏻
